ArduCopter/mode.h

```diff
@@ -548,6 +548,9 @@ class AutoTune : public AC_AutoTune
 
 class ModeAutoTune : public Mode {
 
+    // ParametersG2 sets a pointer within our autotune object:
+    friend class ParametersG2;
```

Comment:
I thought the parameters were already available to modes using copter.g2.  This is what the other modes use so we might want to maintain that pattern.

Reply:
This is the other way around - g2 has to modify a pointer within the private state of ModeAutoTune

Reply:
```
../../ArduCopter/Parameters.cpp: In constructor ‘ParametersG2::ParametersG2()’:
../../ArduCopter/Parameters.cpp:1087:41: error: ‘AutoTune ModeAutoTune::autotune’ is private within this context
     ,autotune_ptr(&copter.mode_autotune.autotune)
                                         ^~~~~~~~
compilation terminated due to -Wfatal-errors.

```